lib/repo_tools.pm

```diff
@@ -120,6 +121,17 @@ sub add_qa_web_repo {
     zypper_ar($repo, name => 'qa-web', no_gpg_check => is_sle("<12") ? 0 : 1) if ($repo);
 }
 
+=head2 get_installed_patterns
+
+ get_installed_patterns();
+
+Returns array containing all installed patterns in the system.
+
+=cut
+sub get_installed_patterns {
+    split(/\s+/, script_output q[zypper -n se -i -t pattern | tail --lines=+6 | awk -F "|" '{print $2}' | xargs]);
```

Comment:
I think what you were looking for was quite option '-q' instead of `tail --lines=+6` which works as far as the package maintainers would not add '|' there and besides that, depending on the scenarios if there is a zypper ref or not some warning could appear (although still would work for that case).

Reply:
Thanks for the hint. It helped me to look at the issue more deeply and I've found even better approach to parse the output of zypper. I've used `-x` flag that makes output in XML format. Now parsing is more precise.

Reply:
that is exactly the way to go whenever the command allow us, instead of parsing the output. nice code.